plugin/hover.py

```diff
@@ -70,10 +71,17 @@ def run(
         point: Optional[int] = None,
         event: Optional[dict] = None
     ) -> None:
-        hover_point = point or self.view.sel()[0].begin()
+        temp_point = point
+        if temp_point is None:
+            region = first_selection(self.view)
+            if region:
+                temp_point = region.begin()
+        if temp_point is None:
+            return
         window = self.view.window()
         if not window:
             return
+        hover_point = temp_point
```

Comment:
Had to wrangle with types here a bit because pyright insisted that `hover_point` can be `None` even after the `return` that returns on `None`.

Reply:
What do you think of this:
```python
    def run(
        self,
        edit: sublime.Edit,
        only_diagnostics: bool = False,
        point: Optional[int] = None,
        event: Optional[dict] = None
    ) -> None:
        window = self.view.window()
        if not window:
            return
        region = first_selection(self.view)
        if point is None and region is None:
            return
        hover_point = point or region.begin()
        wm = windows.lookup(window)
```

Reply:
I find your solution also ok, but just wanted to cut off the indentation.

Reply:
I like it :)

Reply:
mypy doesn't like it though... It thinks that `region` in `region.begin()` can be `None` but it's not true.

Reply:
I opened an issue https://github.com/microsoft/pyright/issues/1788

Reply:
There is no bug in pyright. It's a bug (or limitation) in mypy.